src/UI/Pages/DesiredState/DesiredStates.test.tsx

```diff
@@ -0,0 +1,266 @@
+  const { component, apiHelper } = setup();
+  render(component);
+
+  expect(
+    await screen.findByRole("generic", { name: "DesiredStatesView-Loading" })
+  ).toBeInTheDocument();
+
+  apiHelper.resolve(Either.left("error"));
+
+  expect(
+    await screen.findByRole("generic", { name: "DesiredStatesView-Failed" })
+  ).toBeInTheDocument();
+});
+
+test("AgentsView shows success table", async () => {
+  const { component, apiHelper } = setup();
+  render(component);
+
+  expect(
+    await screen.findByRole("generic", { name: "DesiredStatesView-Loading" })
+  ).toBeInTheDocument();
+
+  apiHelper.resolve(Either.right(DesiredStateVersions.response));
+
+  expect(
+    await screen.findByRole("grid", { name: "DesiredStatesView-Success" })
+  ).toBeInTheDocument();
+});
+
+test("When using the status filter then only the matching desired states should be fetched and shown", async () => {
+  const { component, apiHelper } = setup();
+  render(component);
+
+  await act(async () => {
+    await apiHelper.resolve(Either.right(DesiredStateVersions.response));
+  });
+
+  const initialRows = await screen.findAllByRole("row", {
+    name: "DesiredStates Table Row",
+  });
+  expect(initialRows).toHaveLength(9);
+
+  userEvent.click(
+    within(screen.getByRole("generic", { name: "FilterBar" })).getByRole(
+      "button",
+      { name: "FilterPicker" }
+    )
+  );
+  userEvent.click(screen.getByRole("option", { name: "Status" }));
+
+  const input = screen.getByPlaceholderText("Select status...");
+  userEvent.click(input);
+
+  const option = await screen.findByRole("option", {
+    name: "skipped_candidate",
+  });
+  await userEvent.click(option);
+
+  expect(apiHelper.pendingRequests[0].url).toEqual(
+    `/api/v2/desiredstate?limit=20&sort=version.desc&filter.status=skipped_candidate`
+  );
+
+  await act(async () => {
+    await apiHelper.resolve(
+      Either.right({
+        ...DesiredStateVersions.response,
+        data: DesiredStateVersions.response.data.slice(0, 3),
+      })
+    );
+  });
+
+  const rowsAfter = await screen.findAllByRole("row", {
+    name: "DesiredStates Table Row",
+  });
+  expect(rowsAfter).toHaveLength(3);
+});
+
+it("When using the Date filter then the desired state versions within the range selected range should be fetched and shown", async () => {
+  const { component, apiHelper } = setup();
+  render(component);
+
+  await act(async () => {
+    await apiHelper.resolve(Either.right(DesiredStateVersions.response));
+  });
+
+  const initialRows = await screen.findAllByRole("row", {
+    name: "DesiredStates Table Row",
+  });
+  expect(initialRows).toHaveLength(9);
+
+  userEvent.click(
+    within(screen.getByRole("generic", { name: "FilterBar" })).getByRole(
+      "button",
+      { name: "FilterPicker" }
+    )
+  );
+  userEvent.click(screen.getByRole("option", { name: "Date" }));
+
+  const fromDatePicker = await screen.findByLabelText("From Date Picker");
```

Comment:
Why do we need .find here? Is the date picker not immediately present?
Or does the test have to wait once you have selected the Data filter?